core/auth/auth.go

```diff
@@ -1902,7 +1907,58 @@ func (a *Auth) removeAccountAuthTypeCredential(context storage.TransactionContex
 	return nil
 }
 
-func (a *Auth) deleteAccount(context storage.TransactionContext, account model.Account) error {
+func (a *Auth) deleteAccount(context storage.TransactionContext, account model.Account, fromAppsIDs []string) error {
+	if len(fromAppsIDs) == 0 {
+		return errors.Newf("no apps specified")
+	}
+
+	//check that every passed app is available for the account
+	for _, c := range fromAppsIDs {
+		hasApp := account.HasApp(c)
+		if !hasApp {
+			return errors.Newf("%s does not have %s app", account.ID, c)
+		}
+	}
+
+	//we are sure that all passed apps are available for the account
+	//now we have to decide if we have to remove the while account or just to unattach it from specific apps
+	allAccountApps := account.GetApps()
+	if len(allAccountApps) == len(fromAppsIDs) {
+		//means remove all apps => remove the whole account
+		return a.deleteFullAccount(context, account)
+	} else {
+		//means remove specific apps only, so unattach only them
+		return a.deleteAccountFromApps(context, account, fromAppsIDs)
+	}
+}
+
+func (a *Auth) deleteAccountFromApps(context storage.TransactionContext, account model.Account, fromAppsIDs []string) error {
+	for _, idToDelete := range fromAppsIDs {
+		orgApps, err := a.deleteApps(context, account.OrgAppsMemberships, idToDelete)
+		if err != nil {
+			return errors.WrapErrorAction(logutils.ActionDelete, model.TypeAccount, nil, err)
+		}
+
+		err = a.storage.DeleteAccountOrgAppsMemberships(context, account.ID, orgApps)
+		if err != nil {
+			return errors.WrapErrorAction(logutils.ActionDelete, model.TypeAccount, nil, err)
+		}
+	}
+
+	return nil
+}
+func (a *Auth) deleteApps(context storage.TransactionContext, orgApp []model.OrgAppMembership, fromAppsIDs string) ([]model.OrgAppMembership, error) {
```

Comment:
This function has misleading name. It does not delete any apps but gives memberships by apps ids.